src/topp/ConsensusID.cpp

```diff
@@ -373,6 +380,31 @@ class TOPPConsensusID :
       new_sp.setMetaValue(SE+":precursor_mass_tolerance_unit",sp.precursor_mass_tolerance_ppm  ? "ppm" : "Da");
       new_sp.setMetaValue(SE+":digestion_enzyme",sp.digestion_enzyme.getName());
       new_sp.setMetaValue(SE+":enzyme_term_specificity",EnzymaticDigestion::NamesOfSpecificity[sp.enzyme_term_specificity]);
+      
+      const auto& chg_pair = sp.getChargeRange();
+      if (chg_pair.first != 0 && chg_pair.first < min_chg) min_chg = chg_pair.first;
+      if (chg_pair.second != 0 && chg_pair.second > max_chg) max_chg = chg_pair.second;
+      if (sp.missed_cleavages > mc ) mc = sp.missed_cleavages;
+      if (sp.fragment_mass_tolerance > frag_tol) frag_tol = sp.fragment_mass_tolerance;
+      if (sp.precursor_mass_tolerance > prec_tol) prec_tol = sp.precursor_mass_tolerance;
+      // extends "" to Trypsin and e.g. Trypsin to Trypsin/P
+      if (sp.digestion_enzyme.getName().hasSubstring(enz))
+      {
+        enz = sp.digestion_enzyme.getName();
+      }
+      if (spec.empty())
+      {
+        spec = EnzymaticDigestion::NamesOfSpecificity[sp.enzyme_term_specificity];
+      }
+      //TODO does not handle no-cterm and no-nterm
+      else if (spec == "full" && (sp.enzyme_term_specificity == EnzymaticDigestion::SPEC_SEMI || sp.enzyme_term_specificity == EnzymaticDigestion::SPEC_NONE))
```

Comment:
this is quite hard to read and error prone.
what about a set<enum> just to keep track of what is present.
and at the end, check for presence from most loose setting to strictest once?